src/scrape_config_builder.py

```diff
@@ -79,7 +79,10 @@ def build_probes_scraping_jobs(
                 {"source_labels": ["__address__"], "target_label": "__param_target"},
                 {"source_labels": ["__param_target"], "target_label": "instance"},
                 {"source_labels": ["__param_target"], "target_label": "probe_target"},
-                {"target_label": "__address__", "replacement": external_url.hostname},
+                {
+                    "target_label": "__address__",
+                    "replacement": f"{external_url.hostname}:{external_url.port or ''}",
```

Comment:
If `port` evaluates to false, we'd get `url:` with a stray colon. One option is to use `'80'` (or 443) instead of `''`.

Reply:
Good point, I'll do `':port' or ''`